src/coreclr/vm/field.h

```diff
@@ -235,13 +244,17 @@ class FieldDesc
     }
 
     // Okay, we've stolen too many bits from FieldDescs.  In the RVA case, there's no
-    // reason to believe they will be limited to 22 bits.  So use a sentinel for the
+    // reason to believe they will be limited to .  So use a sentinel for the
     // huge cases, and recover them from metadata on-demand.
     void SetOffsetRVA(DWORD dwOffset)
     {
         LIMITED_METHOD_CONTRACT;
 
-        m_dwOffset = (dwOffset > FIELD_OFFSET_LAST_REAL_OFFSET)
+        // The FIELD_OFFSET_DYNAMIC_RVA is a special case for EnC added fields when the
```

Comment:
It would look better to leave this as is, and introduce `SetDynamicRVA` method instead that sets the special offset. Similar to how `SetEnCNew` is done.

Reply:
That makes things a lot better. Thanks.